source/common/modules/markdown-editor/autocomplete/snippets.ts

```diff
@@ -161,9 +162,16 @@ export const snippetsUpdateField = StateField.define<SnippetStateField>({
     }
 
     // This monstrosity ensures that our ranges stay in sync while the user types
-    val.activeSelections = val.activeSelections.map(selection => {
-      return selection.map(transaction.changes)
-    })
+    val.activeSelections = val.activeSelections
+      .filter(selection => selection.ranges.some(r => {
+        if (transaction.changes.mapPos(r.from, -1, MapMode.TrackBefore) === null) {
```

Comment:
Shouldn't this be [`TrackDel`](https://codemirror.net/docs/ref/#state.MapMode.TrackDel)?

Reply:
Also, I think you can simplify the if/else construct to a direct return